src/test/java/roomescape/reservation/repository/WaitingRepositoryTest.java

```diff
@@ -0,0 +1,117 @@
+import roomescape.reservationtime.domain.ReservationTime;
+import roomescape.reservationtime.repository.ReservationTimeRepository;
+import roomescape.theme.domain.Theme;
+import roomescape.theme.repository.ThemeRepository;
+
+@DataJpaTest
+@Import(TestConfig.class)
+@TestPropertySource(properties = {
+    "spring.sql.init.mode=never"
+})
+class WaitingRepositoryTest {
+
+    private static final LocalDate futureDate = TestFixture.makeFutureDate();
+
+    @Autowired
+    private WaitingRepository waitingRepository;
+
+    @Autowired
+    private MemberRepository memberRepository;
+
+    @Autowired
+    private ReservationTimeRepository reservationTimeRepository;
+
+    @Autowired
+    private ThemeRepository themeRepository;
+
+    private Member member;
+    private ReservationTime time;
+    private Theme theme;
+
+    @BeforeEach
+    void setUp() {
+        member = memberRepository.save(TestFixture.makeMember());
+        time = reservationTimeRepository.save(ReservationTime.withUnassignedId(LocalTime.of(10, 0)));
+        theme = themeRepository.save(TestFixture.makeTheme(1L));
+
+        Waiting w1 = new Waiting(member, new ReservationInfo(futureDate, time, theme),1);
+        waitingRepository.save(w1);
+    }
+
+    @Test
+    void existsByDateAndTimeIdAndThemeId() {
+        boolean exists = waitingRepository.existsByDateAndTimeIdAndThemeId(
+            futureDate, time.getId(), theme.getId()
+        );
+        assertThat(exists).isTrue();
+    }
+
+    @Test
+    void findMaxOrderByDateAndTimeAndTheme() {
+        int initialMax = waitingRepository.findMaxOrderByDateAndTimeAndTheme(
+            futureDate, time.getId(), theme.getId()
+        );
+        assertThat(initialMax).isEqualTo(1);
+
+        Waiting w2 = waitingRepository.save(
+            new Waiting(member, new ReservationInfo(futureDate, time, theme),2)
+        );
+
+        int updatedMax = waitingRepository.findMaxOrderByDateAndTimeAndTheme(
+            futureDate, time.getId(), theme.getId()
+        );
+        assertThat(updatedMax).isEqualTo(2);
+    }
+
+    @Test
+    void findWaitingsWithRankByMemberId() {
+        Waiting w2 = waitingRepository.save(
+            new Waiting(member, new ReservationInfo(futureDate, time, theme),2)
+        );
+
+        List<WaitingWithRank> ranks = waitingRepository.findWaitingsWithRankByMemberId(member.getId());
+        assertThat(ranks).hasSize(2);
+
+        assertThat(ranks)
+            .extracting(WaitingWithRank::getRank)
+            .containsExactly(1L, 2L);
+    }
+
+    @Test
+    void findFirstByInfoDateAndInfoTimeAndInfoThemeOrderByTurnAsc() {
+        Waiting w2 = waitingRepository.save(
+            new Waiting(member, new ReservationInfo(futureDate, time, theme),2)
+        );
+
+        Optional<Waiting> first = waitingRepository
+            .findFirstByInfoDateAndInfoTimeAndInfoThemeOrderByTurnAsc(
+                futureDate, time, theme
+            );
```

Comment:
```suggestion
        waitingRepository.save(new Waiting(member, new ReservationInfo(futureDate, time, theme),2));

        Optional<Waiting> first = waitingRepository
            .findFirstByInfoDateAndInfoTimeAndInfoThemeOrderByTurnAsc(futureDate, time, theme);
```
1. 한줄한줄도 테스트 가독성이므로 불필요한 줄바꿈은 제거해주시면 좋을 것 같아요.
2. 불필요한 변수 선언도 모두 제거해주세요.
3. 여러 assertThat을 사용하는 구분은 assertAll이어야겠네요.
4. `first.get().getTurn()`을 하는데 `isPresent()`를 검증해야하나요?

Reply:
1,2 . 줄바꿈 제거완료했습니다. 
3번의 경우 밑에서 언급하신 불필요한 검증들을 지우다보니 여러 개를 검증할 필요가 없어져 하나의 assertThat()으로 처리완료했습니다!